### PR TITLE
[WIP] Add OnePassword Extension (#299)

### PR DESCRIPTION
**Pull request by huyanhh**
[manual] I'm a little unsure about what the URL is needed to find the Login. For now
I have put '@https:/chat.zulip.org'.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@huyanhh, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by borisyankov** [comment]
You probably want to add the `.tmp' directory to `.gitignore`.

**Comment by huyanhh** [comment]
`.tmp` has been added. Also just reverted the `pbxproj`, removing my person bundle identifier, but I'm not sure about the rest of the file, because it looks like it was modified to link the library together.

File: src/start/PasswordAuthView.js [hunk 1]
```diff
@@ -28,9 +37,31 @@ class PasswordAuthView extends React.Component {
...
+        },
+        () => {
+          console.log('OnePassword not available.');
+          this.setState({ onepasswordHidden: true });
+        }
+      );
+  }
+
+  findOnepasswordLogin = () => {
+    OnePassword.findLogin('@https://chat.zulip.org')
```
Comment (neerajwahi) [inline src/start/PasswordAuthView.js]: For the URL you'll want to pass `realm` as prop to `PasswordAuthView` from `AuthScreen`

File: src/start/PasswordAuthView.js [hunk 2]
```diff
@@ -28,9 +37,31 @@ class PasswordAuthView extends React.Component {
...
+          this.setState({ onepasswordHidden: false });
+        },
+        () => {
+          console.log('OnePassword not available.');
+          this.setState({ onepasswordHidden: true });
+        }
+      );
+  }
+
+  findOnepasswordLogin = () => {
```
Comment (neerajwahi) [inline src/start/PasswordAuthView.js]: Let's change this to `findOnePasswordLogin` just for consistent capitalization

File: src/start/PasswordAuthView.js [hunk 3]
```diff
@@ -20,6 +28,7 @@ class PasswordAuthView extends React.Component {
     email: string,
     password: string,
     error: string,
+    onepasswordHidden: boolean,
```
Comment (neerajwahi) [inline src/start/PasswordAuthView.js]: `onepasswordHidden` -> `onePasswordHidden`

Also, I think it's clearer if you change this to `onePasswordSupported`

File: src/start/PasswordAuthView.js [hunk 4]
```diff
@@ -77,13 +108,19 @@ class PasswordAuthView extends React.Component {
...
-          customStyle={styles.field}
+          customStyle={[styles.field, secondaryStyles.passwordField]}
           placeholder="Password"
           secureTextEntry
           value={password}
           onChangeText={newPassword => this.setState({ password: newPassword })}
           blurOnSubmit={false}
-        />
+        >
+          <OnepasswordButton
```
Comment (neerajwahi) [inline src/start/PasswordAuthView.js]: `onePasswordButton`

File: ios/ZulipMobile.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -575,16 +598,14 @@
 		83CBB9F71A601CBA00E9B192 /* Project object */ = {
 			isa = PBXProject;
 			attributes = {
-				LastUpgradeCheck = 0810;
+				LastUpgradeCheck = 810;
 				ORGANIZATIONNAME = Facebook;
 				TargetAttributes = {
 					00E356ED1AD99517003FC87E = {
 						CreatedOnToolsVersion = 6.2;
+						DevelopmentTeam = Q8R76S2QE8;
```
Comment (neerajwahi) [inline ios/ZulipMobile.xcodeproj/project.pbxproj]: Please remove this line

File: ios/ZulipMobile.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -575,16 +598,14 @@
...
 						TestTargetID = 13B07F861A680F5B00A75B9A;
 					};
-					13B07F861A680F5B00A75B9A = {
```
Comment (neerajwahi) [inline ios/ZulipMobile.xcodeproj/project.pbxproj]: Please add back these lines

File: ios/ZulipMobile.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -940,6 +973,7 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				BUNDLE_LOADER = "$(TEST_HOST)";
+				DEVELOPMENT_TEAM = Q8R76S2QE8;
```
Comment (neerajwahi) [inline ios/ZulipMobile.xcodeproj/project.pbxproj]: Remove

File: ios/ZulipMobile.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -963,13 +998,15 @@
 			buildSettings = {
 				BUNDLE_LOADER = "$(TEST_HOST)";
 				COPY_PHASE_STRIP = NO;
+				DEVELOPMENT_TEAM = Q8R76S2QE8;
```
Comment (neerajwahi) [inline ios/ZulipMobile.xcodeproj/project.pbxproj]: Remove

File: ios/ZulipMobile.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -982,13 +1019,14 @@
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				DEAD_CODE_STRIPPING = NO;
-				DEVELOPMENT_TEAM = 66KHCWMEYB;
+				DEVELOPMENT_TEAM = "";
```
Comment (neerajwahi) [inline ios/ZulipMobile.xcodeproj/project.pbxproj]: Undo change

File: ios/ZulipMobile.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -1007,13 +1045,14 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
-				DEVELOPMENT_TEAM = 66KHCWMEYB;
+				DEVELOPMENT_TEAM = "";
```
Comment (neerajwahi) [inline ios/ZulipMobile.xcodeproj/project.pbxproj]: Undo change

File: src/start/PasswordAuthView.js (hunk 4 above)
Comment (huyanhh) [inline src/start/PasswordAuthView.js]: Do you mean OnePasswordButton, as it is a name of a component?

**Comment by huyanhh** [comment]
Updated with requested changes and also did a little bit of refactoring #da98c5c

**Comment by borisyankov** [comment]
I think this PR needs a bit more work but is helpful to have it.
@huyanhh if you are free to finish it, the efforts will be appreciated 👍

**Comment by borisyankov** [comment]
@huyanhh closing this for now.
Please, if you have time and desire, don't hesitate to open it again and let's work on completing it.

**Pull request closed without merging**